decidim-elections/app/commands/decidim/votings/census/admin/create_datum.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Votings
+    module Census
+      module Admin
+        # A command with the business logic to create the datum for a
+        # dataset row.
+        class CreateDatum < Rectify::Command
+          def initialize(form, dataset)
+            @form = form
+            @dataset = dataset
+          end
+
+          # Executes the command. Broadcast this events:
+          # - :ok when everything is valid
+          # - :invalid when the form wasn't valid and couldn't proceed-
+          #
+          # Returns nothing.
+          def call
+            return broadcast(:invalid) unless form.valid?
+
+            create_census_datum!
+            broadcast(:ok)
+          end
+
+          attr_reader :form, :dataset
+
+          def create_census_datum!
+            attributes = {
+              hashed_in_person_data: form.hashed_in_person_data,
+              hashed_check_data: form.hashed_check_data,
+
+              full_name: form.full_name,
+              full_address: form.full_address,
+              postal_code: form.postal_code,
+              mobile_phone_number: form.mobile_phone_number,
+              email: form.email
+            }
```

Comment:
Not sure about this but I think you can just call `form.attributes` and it will return the same hash that you are building 👍

Reply:
Thanks, is good to know this, but in this case the form has more attributes than the model because they are used to generate the `hashed_*_data` attributes and those are not persisted.